src/tracker_manager.cpp

```diff
@@ -291,7 +292,7 @@ constexpr tracker_request_flags_t tracker_request::i2p;
 #endif
 		{
 			auto con = std::make_shared<http_tracker_connection>(ios, *this, std::move(req), c);
-			if (m_http_conns.size() < std::size_t(sett.get_int(settings_pack::max_concurrent_http_announces)))
+			if (priority || m_http_conns.size() < std::size_t(sett.get_int(settings_pack::max_concurrent_http_announces)))
```

Comment:
this makes me feel a bit uneasy. It effectively removes this cap. How about just using `push_front()` if `priority` is set instead?